// Copyright 2024 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


enum BillingbudgetsAlphaBudgetBudgetFilterCreditTypesTreatmentEnum {
  BillingbudgetsAlphaBudgetBudgetFilterCreditTypesTreatmentEnumNO_VALUE_DO_NOT_USE = 0;
  BillingbudgetsAlphaBudgetBudgetFilterCreditTypesTreatmentEnumINCLUDE_ALL_CREDITS = 1;
  BillingbudgetsAlphaBudgetBudgetFilterCreditTypesTreatmentEnumEXCLUDE_ALL_CREDITS = 2;
  BillingbudgetsAlphaBudgetBudgetFilterCreditTypesTreatmentEnumINCLUDE_SPECIFIED_CREDITS = 3;
}

enum BillingbudgetsAlphaBudgetBudgetFilterCalendarPeriodEnum {
  BillingbudgetsAlphaBudgetBudgetFilterCalendarPeriodEnumNO_VALUE_DO_NOT_USE = 0;
  BillingbudgetsAlphaBudgetBudgetFilterCalendarPeriodEnumCALENDAR_PERIOD_UNSPECIFIED = 1;
  BillingbudgetsAlphaBudgetBudgetFilterCalendarPeriodEnumMONTH = 2;
  BillingbudgetsAlphaBudgetBudgetFilterCalendarPeriodEnumQUARTER = 3;
  BillingbudgetsAlphaBudgetBudgetFilterCalendarPeriodEnumYEAR = 4;
}

enum BillingbudgetsAlphaBudgetThresholdRulesSpendBasisEnum {
  BillingbudgetsAlphaBudgetThresholdRulesSpendBasisEnumNO_VALUE_DO_NOT_USE = 0;
  BillingbudgetsAlphaBudgetThresholdRulesSpendBasisEnumBASIS_UNSPECIFIED = 1;
  BillingbudgetsAlphaBudgetThresholdRulesSpendBasisEnumCURRENT_SPEND = 2;
  BillingbudgetsAlphaBudgetThresholdRulesSpendBasisEnumFORECASTED_SPEND = 3;
}

message BillingbudgetsAlphaBudget {
  string name = 1;
  string display_name = 2;
  BillingbudgetsAlphaBudgetBudgetFilter budget_filter = 3;
  BillingbudgetsAlphaBudgetAmount amount = 4;
  repeated BillingbudgetsAlphaBudgetThresholdRules threshold_rules = 5;
  string etag = 6;
  BillingbudgetsAlphaBudgetAllUpdatesRule all_updates_rule = 7;
  string billing_account = 8;
}

message BillingbudgetsAlphaBudgetBudgetFilter {
  repeated string projects = 1;
  repeated string credit_types = 2;
  BillingbudgetsAlphaBudgetBudgetFilterCreditTypesTreatmentEnum credit_types_treatment = 3;
  repeated string services = 4;
  repeated string subaccounts = 5;
  map<string, BillingbudgetsAlphaBudgetBudgetFilterLabels> labels = 6;
  BillingbudgetsAlphaBudgetBudgetFilterCalendarPeriodEnum calendar_period = 7;
  BillingbudgetsAlphaBudgetBudgetFilterCustomPeriod custom_period = 8;
}

message BillingbudgetsAlphaBudgetBudgetFilterLabels {
  repeated string values = 1;
}

message BillingbudgetsAlphaBudgetBudgetFilterCustomPeriod {
  BillingbudgetsAlphaBudgetBudgetFilterCustomPeriodStartDate start_date = 1;
  BillingbudgetsAlphaBudgetBudgetFilterCustomPeriodEndDate end_date = 2;
}

message BillingbudgetsAlphaBudgetBudgetFilterCustomPeriodStartDate {
  int64 year = 1;
  int64 month = 2;
  int64 day = 3;
}

message BillingbudgetsAlphaBudgetBudgetFilterCustomPeriodEndDate {
  int64 year = 1;
  int64 month = 2;
  int64 day = 3;
}

message BillingbudgetsAlphaBudgetAmount {
  BillingbudgetsAlphaBudgetAmountSpecifiedAmount specified_amount = 1;
  BillingbudgetsAlphaBudgetAmountLastPeriodAmount last_period_amount = 2;
}

message BillingbudgetsAlphaBudgetAmountSpecifiedAmount {
  string currency_code = 1;
  int64 units = 2;
  int64 nanos = 3;
}

message BillingbudgetsAlphaBudgetAmountLastPeriodAmount {
}

message BillingbudgetsAlphaBudgetThresholdRules {
  double threshold_percent = 1;
  BillingbudgetsAlphaBudgetThresholdRulesSpendBasisEnum spend_basis = 2;
}

message BillingbudgetsAlphaBudgetAllUpdatesRule {
  string pubsub_topic = 1;
  string schema_version = 2;
  repeated string monitoring_notification_channels = 3;
  bool disable_default_iam_recipients = 4;
}

message ApplyBillingbudgetsAlphaBudgetRequest {
  BillingbudgetsAlphaBudget resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteBillingbudgetsAlphaBudgetRequest {
  string service_account_file = 1;
  BillingbudgetsAlphaBudget resource = 2;
}

message ListBillingbudgetsAlphaBudgetRequest {
  string service_account_file = 1;
  string BillingAccount = 2;
}

message ListBillingbudgetsAlphaBudgetResponse {
  repeated BillingbudgetsAlphaBudget items = 1;
}

service BillingbudgetsAlphaBudgetService {
  rpc ApplyBillingbudgetsAlphaBudget(ApplyBillingbudgetsAlphaBudgetRequest) returns (BillingbudgetsAlphaBudget);
  rpc DeleteBillingbudgetsAlphaBudget(DeleteBillingbudgetsAlphaBudgetRequest) returns (google.protobuf.Empty);
  rpc ListBillingbudgetsAlphaBudget(ListBillingbudgetsAlphaBudgetRequest) returns (ListBillingbudgetsAlphaBudgetResponse);
}
